templates/commands/render.go

```diff
@@ -322,6 +343,92 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 	return nil
 }
 
+// processUnknownInputs checks for any unknown input flags and returns an error.
+func processUnknownInputs(flagInputs map[string]string, spec *model.Spec) error {
+	var merr error
+
+	specInputs := make(map[string]any)
+	for _, v := range spec.Inputs {
+		specInputs[v.Name.Val] = struct{}{}
+	}
+
+	for key, value := range flagInputs {
+		if _, ok := specInputs[key]; !ok {
+			merr = errors.Join(merr, fmt.Errorf("  - %s=%s", key, value))
+		}
+	}
+
+	return merr
+}
+
+// validateFlagInputs validates the input values.
+func (r *Render) validateFlagInputs(spec *model.Spec) error {
```

Comment:
I'd suggest renaming this function. The current name `validateFlagInputs` makes it sound like this function is read-only, but it mutates the state of `r.inputs` by calling promptForInput.